lib/common/src/Dom/Document.php

```diff
@@ -573,19 +586,54 @@ private function normalizeDocumentStructure($content)
      */
     public function normalizeDomStructure()
     {
-        $head = $this->getElementsByTagName(Tag::HEAD)->item(0);
-        if (! $head) {
-            $this->head = $this->createElement(Tag::HEAD);
-            $this->insertBefore($this->head, $this->firstChild);
```

Comment:
This, like the `body` handling below, appears to have been a bug where it was appending the `head` to the document and not to the `documentElement`.